From ba0c2cd3eed989891667468a34135f4618278f70 Mon Sep 17 00:00:00 2001
From: Steven Barrett <steven@liquorix.net>
Date: Thu, 27 Apr 2023 14:43:57 -0500
Subject: [PATCH 13/14] ZEN: Set default max map count to (INT_MAX - 5)

Per Fedora, they intend to change the default max map count for their
distribution to improve OOTB compatibility with games played through
Steam/Proton.  The value they picked comes from the Steam Deck where
valve defaults to INT_MAX - 5. [1]

Since most Zen Kernel / Liquorix users probably play games, follow
Valve's lead and raise this value to their default.

[1] https://fedoraproject.org/wiki/Changes/IncreaseVmMaxMapCount
---
 include/linux/mm.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/mm.h b/include/linux/mm.h
index bf5d0b1b1..c2f0a8a60 100644
--- a/include/linux/mm.h
+++ b/include/linux/mm.h
@@ -191,7 +191,7 @@ static inline void __mm_zero_struct_page(struct page *page)
  * that.
  */
 #define MAPCOUNT_ELF_CORE_MARGIN	(5)
-#define DEFAULT_MAX_MAP_COUNT	(USHRT_MAX - MAPCOUNT_ELF_CORE_MARGIN)
+#define DEFAULT_MAX_MAP_COUNT	(INT_MAX - 5)
 
 extern int sysctl_max_map_count;
 
-- 
2.42.0

